test/kaniko_task_test.go

```diff
@@ -0,0 +1,185 @@
+// +build e2e
+
+/*
+Copyright 2018 Knative Authors LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	knativetest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+
+	// Mysteriously by k8s libs, or they fail to create `KubeClient`s from config. Apparently just importing it is enough. @_@ side effects @_@. https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	kanikoTaskName     = "kanikotask"
+	kanikoTaskRunName  = "kanikotask-run"
+	kanikoResourceName = "go-example-git"
+	kanikoBuildOutput  = "Build successful"
+)
+
+func getGitResource(namespace string) *v1alpha1.PipelineResource {
+	return &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kanikoResourceName,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: v1alpha1.PipelineResourceTypeGit,
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/pivotal-nader-ziada/gohelloworld",
+				},
+			},
+		},
+	}
+}
+
+func getTask(namespace string, t *testing.T) *v1alpha1.Task {
+	dockerRepo := os.Getenv("DOCKER_REPO_OVERRIDE")
```

Comment:
maybe put a comment here just explaining briefly (as much as we know) where this env var is expected to come from?

also maybe if `DOCKER_REPO_OVERRIDE` isn't provided, we should default to `KO_DOCKER_REPO`? our [environment setup](https://github.com/knative/build-pipeline/blob/master/DEVELOPMENT.md#environment-setup) and [our integration test docs](https://github.com/knative/build-pipeline/blob/master/test/README.md#integration-tests) don't refer to it, and I think we should avoiding introducing it if we can

Reply:
I used this one because test-infra says it will be set in the cluster: https://github.com/knative/test-infra/blob/13055d769cc5e1756e605fcb3bcc1c25376699f1/scripts/README.md

Reply:
what if you're running it directly with `go test`? in that case the dev would have to provide `DOCKER_REPO_OVERRIDE` but they probably would already have `KO_DOCKER_REPO` set, so I'd really rather use `KO_DOCKER_REPO` in that case, v.s updating our docs to tell devs on this project to set both

Reply:
ok, I will change to using `KO_DOCKER_REPO`